bug: implicit enum selector in struct field default emits empty C expression

## Description

When a struct field default value uses an implicit enum selector (`.VARIANT`), the typechecker accepts it but codegen emits an empty expression, causing a C compiler error to leak through.

Explicit selectors (`EnumName.VARIANT`) work correctly in the same position.

## Minimal Reproduction

```grayscale
const Status enum {
    PENDING
    ACTIVE
}

const Task struct {
    status Status = .PENDING
}

do main() {
    mut t = Task{}
    println("${t.status}")
}
```

## Expected Behavior

Compiles and prints `0` (the integer value of `Status.PENDING`), identical to using `status Status = Status.PENDING`.

## Actual Behavior

C compilation error leaks through:

```
/tmp/gray_probe26.c:27:53: error: expected expression
   27 |     GrayStruct_Task t = (GrayStruct_Task){.status = };
                                                            ^
```

The implicit selector `.PENDING` emits nothing in the generated C struct literal initializer.

## Workaround

Use the explicit form: `status Status = Status.PENDING`

## Notes

- Implicit selectors work correctly in all other contexts: function arguments, assignments, comparisons, `when`/`is` branches, struct literal fields, and array literals.
- This is the classic "typechecker accepts it, codegen crashes" pattern — the fix likely belongs in codegen where struct default field values are emitted.

--- a/grayc/src/typechecker/typechecker.c
+++ b/grayc/src/typechecker/typechecker.c
@@ -11131,6 +11131,14 @@ static void register_declarations(TypeChecker *checker, AstNode *program) {
                         break;
                     }
                 }
+                /* Resolve implicit enum selectors in field default values */
+                if (stmt->data.struct_decl.fields[j].default_value) {
+                    GrayType *saved_expected = checker->expected_type;
+                    if (ftypes[j] && ftypes[j]->kind == TK_ENUM && ftypes[j]->name)
+                        checker->expected_type = ftypes[j];
+                    resolve_expression(checker, stmt->data.struct_decl.fields[j].default_value);
+                    checker->expected_type = saved_expected;
+                }
             }
             /* E2037/E2038: reserved name check for structs */
             const char *sn = STRUCT_DISPLAY_NAME(stmt);

--- a/integration-tests/pass/core/implicit_enum_struct_field_default.gray
+++ b/integration-tests/pass/core/implicit_enum_struct_field_default.gray
@@ -1,0 +1,79 @@
+/*
+ * Pass Test: Implicit enum selector in struct field default values
+ * Tests that .VARIANT syntax works in struct field defaults.
+ */
+
+const Status enum {
+    PENDING
+    ACTIVE
+    DONE
+}
+
+const Task struct {
+    status Status = .PENDING
+}
+
+const Job struct {
+    state Status = .ACTIVE
+    name string = "untitled"
+}
+
+do main() {
+    mut passed int = 0
+    mut failed int = 0
+
+    // Test 1: implicit enum default used via empty literal
+    mut t = Task{}
+    if t.status == Status.PENDING {
+        println("  [PASS] Task{}.status == PENDING")
+        passed += 1
+    } otherwise {
+        println("  [FAIL] Task{}.status: expected PENDING")
+        failed += 1
+    }
+
+    // Test 2: implicit enum default used via new()
+    mut t2 = new(Task)
+    if t2^.status == Status.PENDING {
+        println("  [PASS] new(Task).status == PENDING")
+        passed += 1
+    } otherwise {
+        println("  [FAIL] new(Task).status: expected PENDING")
+        failed += 1
+    }
+
+    // Test 3: override implicit enum default
+    mut t3 = Task{status: .DONE}
+    if t3.status == Status.DONE {
+        println("  [PASS] Task{status: .DONE}.status == DONE")
+        passed += 1
+    } otherwise {
+        println("  [FAIL] Task{status: .DONE}: expected DONE")
+        failed += 1
+    }
+
+    // Test 4: second field with implicit enum default
+    mut j = Job{}
+    if j.state == Status.ACTIVE {
+        println("  [PASS] Job{}.state == ACTIVE")
+        passed += 1
+    } otherwise {
+        println("  [FAIL] Job{}.state: expected ACTIVE")
+        failed += 1
+    }
+
+    if j.name == "untitled" {
+        println("  [PASS] Job{}.name == \"untitled\"")
+        passed += 1
+    } otherwise {
+        println("  [FAIL] Job{}.name: expected \"untitled\"")
+        failed += 1
+    }
+
+    println("Results: ${passed} passed, ${failed} failed")
+    if failed > 0 {
+        println("SOME TESTS FAILED")
+    } otherwise {
+        println("ALL TESTS PASSED")
+    }
+}

